src/main/webapp/app/exam/manage/exams/exam-update.component.ts

```diff
@@ -44,99 +43,163 @@ export class ExamUpdateComponent implements OnInit {
+            if (!exam.gracePeriod) {
+                exam.gracePeriod = 180;
             }
 
-            this.courseManagementService.find(Number(this.route.snapshot.paramMap.get('courseId'))).subscribe({
-                next: (response: HttpResponse<Course>) => {
-                    this.exam.course = response.body!;
-                    this.course = response.body!;
-                    this.hideChannelNameInput = (exam.id !== undefined && exam.channelName === undefined) || !isMessagingEnabled(this.course);
-                },
-                error: (err: HttpErrorResponse) => onError(this.alertService, err),
-            });
-
-            if (!this.exam.gracePeriod) {
-                this.exam.gracePeriod = 180;
-            }
             // test exam only feature automatic assessment
-            if (this.exam.testExam) {
-                this.exam.numberOfCorrectionRoundsInExam = 0;
-            } else if (!this.exam.numberOfCorrectionRoundsInExam) {
-                this.exam.numberOfCorrectionRoundsInExam = 1;
+            if (exam.testExam) {
+                exam.numberOfCorrectionRoundsInExam = 0;
+            } else if (!exam.numberOfCorrectionRoundsInExam) {
+                exam.numberOfCorrectionRoundsInExam = 1;
             }
+
+            this.exam = exam;
+            this.isImport = isImport;
+            this.isImportInSameCourse = isImport && exam.course?.id === data.course.id;
+            this.originalStartDate = exam.startDate?.clone();
+            this.originalEndDate = exam.endDate?.clone();
+
+            this.course = data.course;
+            this.exam.course = data.course;
+            this.hideChannelNameInput = (!!exam.id && !exam.channelName) || !isMessagingEnabled(this.course);
         });
-        // Initialize helper attributes
-        this.workingTimeInMinutes = this.exam.workingTime! / 60;
-        this.calculateMaxWorkingTime();
+    }
+
+    /**
+     * Sets the exam working time in minutes.
+     * @param minutes
+     */
+    set workingTimeInMinutes(minutes: number) {
+        this.exam.workingTime = minutes * 60;
+    }
+
+    /**
+     * Returns the exam working time in minutes.
+     */
+    get workingTimeInMinutes(): number {
+        return this.exam.workingTime ? this.exam.workingTime / 60 : 0;
     }
 
     /**
      * Revert to the previous state, equivalent with pressing the back button on your browser
      * Returns to the detail page if there is no previous state and we edited an existing exam
      * Returns to the overview page if there is no previous state and we created a new exam
      */
-    previousState() {
+    resetToPreviousState() {
         this.navigationUtilService.navigateBackWithOptional(['course-management', this.course.id!.toString(), 'exams'], this.exam.id?.toString());
     }
 
+    /**
+     * Calculates the WorkingTime for real exams based on the start- and end-time.
+     */
+    calculateWorkingTime() {
```

Comment:
There should already be a method computing this

Reply:
I mean it was there before, or do you mean a re-usable method somewhere else in the codebase?

Reply:
Used `normalWorkingTime` :)